### PR TITLE
Safeguard in DL1 data check against empty muon files + solved pyplot related issue

### PR DESCRIPTION
**Pull request by moralejo**
Solves issues occurring when muon files have no entries, as well as a strange pyplot.hist2d error, which failed very few times (I don't know why) when inputs were pandas data series (instead of numpy arrays).

This hopefully addresses issues #414 and #416. Tested it on some of the failing files, I think it should work for all of them.

### CODE REVIEW

File: lstchain/datachecks/containers.py [hunk 1]
```diff
@@ -173,35 +173,38 @@ def fill_event_wise_info(self, subrun_index, table, mask, geom,
...
-                                     table['skewness'][mask],
+        counts, _, _, _ = plt.hist2d(intensity,
+                                     np.array(table['skewness'][mask]),
                                      bins=histogram_binnings.hist_skewness)
         self.hist_skewness = counts
 
-        intercept = table['intercept'][mask]
         counts, _, _, _ = \
-            plt.hist2d(intensity, intercept,
+            plt.hist2d(intensity, np.array(table['intercept'][mask]),
```
Comment (MaxNoe) [inline lstchain/datachecks/containers.py:190]: is tables a pandas series? if yes, `table.loc[mask, 'intercept'].to_numpy()` should be more efficient.

File: lstchain/datachecks/containers.py [hunk 2]
```diff
@@ -173,35 +173,38 @@ def fill_event_wise_info(self, subrun_index, table, mask, geom,
...
                        bins=histogram_binnings.hist_tgrad_vs_length)
         self.hist_tgrad_vs_length = counts
+
+        # We noticed an occasional pyplot error that seems to be fixed by
+        # making sure that the coordinates passed to hist2d are ndarrays
+        # (instead of Pandas data series)
+
         counts, _, _, _ = \
-            plt.hist2d(length[intensity > 200], tgrad[intensity > 200],
+            plt.hist2d(np.array(length[intensity > 200]),
```
Comment (MaxNoe) [inline lstchain/datachecks/containers.py:206]: also here, `.to_numpy()` instead of np.array

Comment (moralejo) [reply lstchain/datachecks/containers.py]: Ok, done!

**Comment by morcuended** [comment]
Tested on real data. With these changes, those errors/warnings reported on #414 and #416 do not raise any more.

**Pull request merged**